scripts/symbolicate-profile.ts

```diff
@@ -0,0 +1,208 @@
+ * 2. The profiles created by the app on the user's device have the app version encoded in the filename.
+ * 3. This script takes in a .cpuprofile file, reads the app version from the filename, and downloads the corresponding source map from the artifacts using github's API.
+ * 4. It then uses the source map to symbolicate the .cpuprofile file using the `react-native-release-profiler` cli.
+ *
+ * @note For downloading an artefact a github token is required.
+ */
+import {Octokit} from '@octokit/core';
+import {execSync} from 'child_process';
+import fs from 'fs';
+import https from 'https';
+import path from 'path';
+import * as Logger from './utils/Logger';
+import parseCommandLineArguments from './utils/parseCommandLineArguments';
+
+const argsMap = parseCommandLineArguments();
+
+// #region Input validation
+if (Object.keys(argsMap).length === 0 || argsMap.help !== undefined) {
+    Logger.log('Symbolicates a .cpuprofile file obtained from a specific app version by downloading the source map from the github action runs.');
+    Logger.log('Usage: npm run symbolicate-profile -- --profile=<filename> --platform=<ios|android>');
+    Logger.log('Options:');
+    Logger.log('  --profile=<filename>          The .cpuprofile file to symbolicate');
+    Logger.log('  --platform=<ios|android>      The platform for which the source map was uploaded');
+    Logger.log('  --gh-token                    Token to use for requests send to the GitHub API. By default tries to pick up from the environment variable GITHUB_TOKEN');
+    Logger.log('  --help                        Display this help message');
+    process.exit(0);
+}
+
+if (argsMap.profile === undefined) {
+    Logger.error('Please specify the .cpuprofile file to symbolicate using --profile=<filename>');
+    process.exit(1);
+}
+if (!fs.existsSync(argsMap.profile)) {
+    Logger.error(`File ${argsMap.profile} does not exist.`);
+    process.exit(1);
+}
+
+if (argsMap.platform === undefined) {
+    Logger.error('Please specify the platform using --platform=ios or --platform=android');
+    process.exit(1);
+}
+
+const githubToken = argsMap.ghToken ?? process.env.GITHUB_TOKEN;
+if (githubToken === undefined) {
+    Logger.error('No GitHub token provided. Either set a GITHUB_TOKEN environment variable or pass it using --gh-token');
+    process.exit(1);
+}
+// #endregion
+
+// #region Get the app version
+
+// Formatted as "Profile_trace_for_1.4.81-9.cpuprofile"
+const appVersionRegex = /\d+\.\d+\.\d+(-\d+)?/;
+const appVersion = argsMap.profile.match(appVersionRegex)?.[0];
+if (appVersion === undefined) {
+    Logger.error('Could not extract the app version from the profile filename.');
+    process.exit(1);
+}
+Logger.info(`Found app version ${appVersion} in the profile filename`);
+// #endregion
+
+// #region Utility functions
+// We need the token for the download step
+const octokit = new Octokit({auth: githubToken});
```

Comment:
*suggestion:* Update this script to use `GithubUtils.octokit` instead. You'd need to update the internals slightly in order to [supply the token from a source other than GitHub Actions inputs](https://github.com/Expensify/App/blob/1f54177b08daf1b3783ccbaeadd8a9e188e7c0c6/.github/libs/GithubUtils.ts#L74), but then I think this would be simpler and more consistent.

For example, instead of using `request`, you can use `octokit.downloadArtifact`.